netket/utils/group/_group.py

```diff
@@ -263,50 +264,79 @@ def character_table_readable(self) -> Tuple[List[str], Array]:
         return representatives, self.character_table_by_class
 
     @struct.property_cached
-    def _irrep_matrices(self) -> PyTree:
-        random = np.random.standard_normal
-        # Generate the eigensystem of a random linear combination of matrices
-        # in the group's regular representation (namely, product_table == g)
-        # The regular representation contains d copies of each d-dimensional irrep
-        # all of which return the same eigenvalues with eigenvectors that
-        # correspond to one another in the bases of the several irrep copies
-        e, vs = np.linalg.eig(random(len(self))[self.product_table])
-        e = np.stack((e.real, e.imag))
-        # we only need one eigenvector per eigenvalue
-        _, idx = np.unique(comparable(e), return_index=True, axis=1)
-        vs = vs[:, idx]
-        # We will nedd the true product table as well
+    def _irrep_matrices(self, key=jax.random.PRNGKey(0)) -> List[Array]:
+        """
+        Generates irrep matrices using Dixon's algorithm (Math. Comp. 24 (1970), 707).
+
+        We decompose the regular representation of the group into its irreps
+        For real irreps, we make sure the matrices are real
+        """
+
+        # Check Frobenius-Schur indicators of all irreps
         true_product_table = self.product_table[self.inverse]
+        inverted_product_table = true_product_table[:, self.inverse]
+        squares = np.diag(true_product_table)
+        frob = np.array(
+            np.rint(
+                np.sum(self.character_table()[:, squares], axis=1).real / len(self)
+            ),
+            dtype=int,
+        )
+
+        def random(n, key, dtype=float):
+            return np.asarray(jax.random.normal(key, (n,), dtype))
```

Comment:
I don't understand why you want to use jax if then you reconvert to numpy

Reply:
because the JAX PRNG is more easily reproducible

Reply:
to make this code less confusing, if this is a numpy code, I would use numpy new random number generator interface which is essentially the same as jax PRNG system.
In the not-so-distant future, Numba will also support those and so we could jit everything.

```python
rng = numpy.random.default_rng(SEED)
out = rng.normal((n,), dtype=dtype)
```

This is perfectly reproducible.

cc @femtobit, what do you think?

Reply:
I'm happy with the current state, but I don't mind either way. Note that in case we want to numba-jit this later we could also just change it then, it's trivial to do after all.

Reply:
It's not entirely necessary, but I'd prefer not to have jax if it's not needed.
I fear that we won't remember why we're using it and it's just confusing.

Reply:
I actually agree, I think that this part on the symmetries is somehow self-contained and should not depend too much on the autodiff things/jax